### PR TITLE
adding openapi document

### PR DESCRIPTION
**Pull request by AND-JAshcroft**

### CODE REVIEW

File: gateway/openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+schemes:
+  - https
+securityDefinitions:
+  UserSecurity:
+    type: basic
+security:
+  - UserSecurity: []
+paths:
+  /v1/unit_data:
+    post:
```
Comment (IsaiahLai111) [inline gateway/openapi.yaml:15]: I think the method should be get at the unit data endpoint

File: gateway/openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+          in: query
+          type: string
+          required: true
+      responses:
+        200:
+          description: Succesfully retrieved schema
+          schema:
+            type: object
+            schema: "#/definitions/sds_schema"
+        404:
```
Comment (IsaiahLai111) [inline gateway/openapi.yaml:101]: This endpoint will also trigger a 400 response, for example when providing an invalid version

File: gateway/openapi.yaml (hunk 1 above)
Comment (AND-JAshcroft) [reply gateway/openapi.yaml:15]: Nice spot will change

File: gateway/openapi.yaml (hunk 2 above)
Comment (AND-JAshcroft) [reply gateway/openapi.yaml:101]: Thanks will just add that in now

Comment (IsaiahLai111) [reply gateway/openapi.yaml:101]: The status code should be 400

File: gateway/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+  - https
+securityDefinitions:
+  UserSecurity:
+    type: basic
+security:
+  - UserSecurity: []
+paths:
+  /v1/unit_data:
+    get:
+      description: GET method that returns the unit suplementary data corresponding to the given dataset_id and unit_id.
```
Comment (JackRimmerAND) [inline gateway/openapi.yaml:16]: Typo here: supplementary instead of suplementary.

File: gateway/openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+          type: string
+          x-example: 5eb3a2a9-9d07-47a9-9df7-0425ed289060
+        - name: unit_id
+          in: query
+          required: true
+          type: string
+          x-example: 43552 
+      responses:
+        200:
+          description: Succesfully retrieved unit suplementary data
```
Comment (JackRimmerAND) [inline gateway/openapi.yaml:30]: Same supplementary typo

File: gateway/openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+              message:
+                type: string
+                example: Unable to process request
+            required:
+              - status
+              - message
+
+  /v1/dataset_metadata:
+    get:
+      description: GET method that returns the dataset metadata corresponding to the given survey_id adn period_id.
```
Comment (JackRimmerAND) [inline gateway/openapi.yaml:271]: Typo here, adn instead of and

**Review by JackRimmerAND — APPROVED** [review]
Incredibly minor typos but apart from that looks fine. Everything seems to work properly when copying into the swagger editor, nice work

**Review by tidmanmike — APPROVED** [review]

**Pull request merged**